### ISSUE TITLE
Can't display Advanced Custom Fields (ACF)

### ISSUE THREAD

**Issue by blakefosdick** [dudaster/ele-custom-skin#114]
When I design my template (previewing with an ACF post example) all looks good. But when I go to put my template in the page I want to display an archive in, the column block is there, the button and icons are there, but non of the fields display. If I change the template to include a regular post field then that displays fine in the page, just none of the ACF's

**Comment by dudaster** [comment]
This is a serious issue. Can you porvide me with a link?

**Comment by blakefosdick** [comment]
Sure here you go... https://midwest.prosanctity.ourladyoftrust.com/

**Comment by dudaster** [comment]
what did you selected on posts widget query? can you provide a screenshot?

**Comment by blakefosdick** [comment]
Thank you Dudaster, user error... For those who might make my same mistake. After your create your Loop theme and are on the second step of placing the "Posts" widget on the page you want the archive posts populated you need to select your ACF group in the Query. 
![image](https://user-images.githubusercontent.com/8249352/70341684-20b1e580-1819-11ea-8e36-97aac00f7112.png)

**Comment by geraeta** [comment]
Hello, I have the same issue! Is there a solution yet? If I create the actual Loop Template in Elementor it works just fine but once I want to use the layout for a post widget it doesn't show up! Thanks.

Using Elementor Pro Version 2.7.3
and Ele Custom Skin Version 1.3.9

**Comment by dudaster** [comment]
There isn’t an issue with it. Maybe your case is special. Please provide a link to take a look.

**Comment by geraeta** [comment]
The website is not public yet. Can I send you an login privately?

**Comment by dudaster** [comment]
If you select classic skin it works?

**Comment by geraeta** [comment]
Yes it can read the post image and post title but none of my costume created ACF Fields

**Comment by geraeta** [comment]
This is the preview when i create the loop:
<img width="1440" alt="Bildschirmfoto 2019-12-14 um 16 07 04" src="https://user-images.githubusercontent.com/58885355/70851338-d6b9a700-1e8b-11ea-895a-51fd460c33b5.png">

and once I want to use it as a skin it doesn't show anything
<img width="1440" alt="Bildschirmfoto 2019-12-14 um 16 07 21" src="https://user-images.githubusercontent.com/58885355/70851347-f9e45680-1e8b-11ea-9fe5-64b6b7bb7684.png">

**Comment by dudaster** [comment]
Make me an user to take a look. Go disaster.com and will take it from chat

**Comment by dudaster** [comment]
The issue seems to be with Advanced Custom Fields: Extended . Deactivate it and it will work properly. I'll look into to see why is this happening.

**Comment by revr0t** [comment]
Hi @dudaster I am having the exact same issue with Advanced Custom Fields: Extended 
Any workaround?

**Comment by dudaster** [comment]
I’m working on a fix. Meanwhile you can write an issue with them also.

**Comment by ds-codes** [comment]
Raised the issue on ACF Extended support forum:
https://wordpress.org/support/topic/incompatibility-with-ele-custom-skin-plugin/

**Comment by acf-extended** [comment]
Hello everyone, hi @dudaster, my respects. I've been sent here by @ds-codes.

I'm not a user of Elementor, but I'll try to replicate the problem to fix it. If I understand correctly, here are the specs:

- WordPress 5.3.2
- Hello Elementor 2.2.2
- Elementor Pro 2.8.3
- ACF PRO 5.8.7
- ACF Extended 0.8.3.1
- Ele Custom Skin 1.3.9

Can someone please write a quick step-by-step setup that doesn't work as intended?
Example: Create new page, Open Elementor, Add ACF widget XXX, configure "Skin" etc...?

Thanks in advance!

Regards.

**Comment by dudaster** [comment]
Hello!

I think you need to follow this steps:

1. Create some posts and add some custom fields to them any kind
2. Create an Elementor Loop Template under the theme builder. see more info here(https://www.youtube.com/watch?v=DwLFdaZ69KU)
3. Add some dynamic content like a Heading Widget and select dynamic value with an ACF field, select the key and then save.
4. Save the Loop template
5. Create a page and add a Posts Widget in that page. Select Custom from Skin field and after that select the Loop template you just made.
6. Save and preview.

What happens behind the hood:

For each post in the loop the template is generated like this:

```
function render_loop_item(){
  global $wp_query;
  $main_query=$wp_query;
  $wp_query = new \WP_Query( array( 'p' => get_the_ID() ) );
  generate_elementor_template($template_id);
  $wp_query = $main_query;
}
```

**Comment by dudaster** [comment]
Hopefully that makes sense to you. If you have any questions don't hesitate to ask!

**Comment by ajit-soni** [comment]
After upgrading the wordpress version to 5.3.2, the Advanced Custom Fields PRO does not appear on the wordpress edit page. Does it support the wordpress version 5.3.2 ?

Please Advise. Thanks!

**Comment by dudaster** [comment]
Please ask on ACF support! But you can try to save the posts with the specific category you want and edit it again and see if it works.

**Comment by ajit-soni** [comment]
Thanks dudaster! That seems to be working but we dont see Advanced Custom Fields: WP WYSIWYG working it to see it as a editor on 5.3.2 ? 

Please Advise. Thanks! 
![screenshot](https://user-images.githubusercontent.com/18722030/71689204-756c6000-2dc7-11ea-8e57-60c63703812d.jpg)

**Comment by revr0t** [comment]
> Hello everyone, hi @dudaster, my respects. I've been sent here by @ds-codes.
> 
> I'm not a user of Elementor, but I'll try to replicate the problem to fix it. If I understand correctly, here are the specs:
> 
> * WordPress 5.3.2
> * Hello Elementor 2.2.2
> * Elementor Pro 2.8.3
> * ACF PRO 5.8.7
> * ACF Extended 0.8.3.1
> * Ele Custom Skin 1.3.9
> 
> Can someone please write a quick step-by-step setup that doesn't work as intended?
> Example: Create new page, Open Elementor, Add ACF widget XXX, configure "Skin" etc...?
> 
> Thanks in advance!
> 
> Regards.

Hi @acf-extended 
Just wanted to touch bases with you again and see if you were able to recreate the issue?
If it is of any help, I have created a WordPress install to demonstrate the issue and can give you admin access? If you like we could also walk you through the setup?

**Comment by dudaster** [comment]
@revr0t do you have a fresh install and you can reproduce this issue?

**Comment by revr0t** [comment]
Hi @dudaster yes, I have uploaded a copy of a project I am working on that has this exact issue. Would you also like admin access?

**Comment by dudaster** [comment]
Please!

**Comment by revr0t** [comment]
@dudaster Just sent you an email with the link and login details.

**Comment by dudaster** [comment]
It seems it works on regular posts it has issues with custom posts.

**Comment by dudaster** [comment]
@revr0t  I think I solved the problem. Please make some tests to see if it's ok.

I've edited to code of the ECS plugin, you can copy it to live to see if it works.

Tomorrow I'll release the update.

**Comment by dudaster** [comment]
@acf-extended it seems if I wp_query a custom post type only by id, ACF with ACF extended doesn't provide the custom field values. 

If you add the post_type to the query seems to fix the issue.

Hope it gives you some clues for future issues that may appear regarding this issue.

**Comment by revr0t** [comment]
@dudaster I just tested it on the first website and it works a dream, no issue.

**Comment by dudaster** [comment]
Sweet I'll roll out an update right now.

**Comment by dudaster** [comment]
Update rolled out.

**Comment by ds-codes** [comment]
Great stuff, @dudaster ! Haven't tried it out yet but that's great support, thank you!

**Comment by revr0t** [comment]
@dudaster thank you again for the 5star support. As a small gesture of my appreciation, I have left a plugin review on the WordPress page. https://wordpress.org/support/plugin/ele-custom-skin/reviews/

**Comment by dudaster** [comment]
You're most kind!

**Comment by acf-extended** [comment]
Hello everyone!

Okay, I'm really sorry for the late answer, I haven't had the time to dive in this ticket earlier. I'm glad to hear that @dudaster found a fix!

I would like to add my input with some precisions on the issue tho, as I believe it is not related to ACF neither to ACF Extended.

I checked the fix in `ele-custom-skin\skins\skin-custom.php:140`, and ran some tests with the original query (before the fix) without ACF & ACF Extended. And yeah the query does not work with pages/custom post types:

`$new_query=new \WP_Query( array( 'p' => get_the_ID() ) );`

This behavior is expected, as documented in the [WP_Query](https://developer.wordpress.org/reference/classes/wp_query/#post-page-parameters) class:

> Remember that default `post_type` is only set to display posts but not pages.

If we don't provide a `post_type` argument in the query, we won't be able to query a specific ID for a page/custom post type (the exception is if you have a `tax_query`, then `post_type` is set to `any` by default).

I don't know why the problem became prominent with ACF Extended, but the query shouldn't even work beforehand.

Note: I ran the test on a blank WordPress setup, without plugins. Code inside a page template:

```
<?php
$query = new WP_Query(array(
    'p' => 4 // ID of a page
));
?>
<pre><?php print_r($query->posts); ?></pre>

<!-- Result: Array() -->
```

Note: I really would like to make it clear if the issue is tied to ACF Extended or not, because I don't want to mess with users queries (and I don't think I do, because I never received any support ticket about such a problem).
